content/300-guides/500-other/880-troubleshooting-orm/100-help-articles/600-vercel-caching-issue.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: 'Vercel deployment caching workaround'
+metaTitle: 'Learn to configure your build process on Vercel to avoid caching-related problems'
+metaDescription: 'Learn to configure your build process on Vercel to avoid caching-related problems'
+---
+
+## Problem
+
+If you deploy an application using Prisma to [Vercel](https://vercel.com/), you may run into strange issues as you iterate on your Prisma schema and deploy updates to your app.
+
+These issues occur because of the fact that Vercel caches the dependencies of your project until one of those dependencies changes. It does this to allow faster builds, and while this is typically a good thing, it causes some problems for Prisma Client.
+
+Prisma uses a `postinstall` hook to generate Prisma Client when dependencies are installed. Because Vercel uses cached modules, this `postinstall` hook never gets run during a deployment after the initial deployment. This results in Prisma Client becoming out of sync with your database schema.
+
+## Solution
+
+This issue can be solved by explicitly generating Prisma Client on every deployment. Running `npx prisma generate` before each deployment will ensure Prisma Client is up-to-date.
+
+You can configure the deployment to run this command in a few different places:
+
+### The application's `build` script in `package.json`
+
+<Admonition type="info">
+
+This is the preferred method as it is a universal solution.
+
+</Admonition>
+
+Within the `scripts` section of your project's `package.json` file, if there is not already a script named `build`, add one and run `npx prisma generate` in that script:
+
+```json highlight=4;add
+{
+  ...
+  "scripts" {
+    "build": "npx prisma generate"
+  }
+  ...
+}
+```
+
+<Admonition type="info">
+
+If you already have a `build` script with existing commands, you can append the `prisma generate` by running `npx prisma generate && <your-existing-scripts>`.
```

Comment:
```suggestion
If you already have a `build` script with existing commands, you can prepend the `prisma generate` by running `npx prisma generate && <your-existing-scripts>`.
```
"running" indicates running some command right now though, to prepend it in the package.json. I think this could be rewritten to make clearer that this means adding something to the existing command.